### PR TITLE
Need a Real Url Type

### PR DESCRIPTION
**Pull request by neonphog**
In starting work on the transport module, I've discovered that we need a real Url type. String isn't gonna cut it anymore : )

### CODE REVIEW

File: crates/core/src/factories/mem_peer_store/test.rs [hunk 1]
```diff
@@ -10,14 +10,22 @@ const AGENT_1: AgentId = AgentId(Id(bytes::Bytes::from_static(b"agent1")));
 const AGENT_2: AgentId = AgentId(Id(bytes::Bytes::from_static(b"agent2")));
 const SPACE_1: SpaceId = SpaceId(Id(bytes::Bytes::from_static(b"space1")));
 
+fn sneak_url(s: &str) -> Url {
```
Comment (ThetaSinner) [inline crates/core/src/factories/mem_peer_store/test.rs:13]: Not following these function names?

Comment (neonphog) [reply crates/core/src/factories/mem_peer_store/test.rs:13]: hehe. the test needs to sneak in some debug data to tally up which infos are still around or whatever, so I'm putting that data into the url and extracting it again. I can add some docs there.

Comment (neonphog) [reply crates/core/src/factories/mem_peer_store/test.rs:16]: Done: https://github.com/holochain/kitsune2/pull/44/commits/81dcec59fe12ab9b915e59646055d70df8abd8f5

**Review by ThetaSinner — APPROVED** [review]

File: crates/api/src/url.rs [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    type Err = K2Error;
+
+    fn from_str(src: &str) -> Result<Self, Self::Err> {
+        Self::from_str(src)
+    }
+}
+
+impl Url {
+    /// Construct a new validated Kitsune2 Url.
+    pub fn new(src: bytes::Bytes) -> K2Result<Self> {
```
[the remark this answers is not in the document]
Comment (neonphog) [reply crates/api/src/url.rs:109]: I'd prefer not... but I guess I don't care too much if you'd rather.

**Review by jost-s — DISMISSED** [review]
Nah, that's fine.

**Review by ThetaSinner — APPROVED** [review]

**Pull request merged**